### PR TITLE
Feat/ufrontend env apps

### PR DESCRIPTION
**Pull request by cbourgois**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/ufrontend-environment`
| Bug fix?         | yes
| New feature?     | yes
| Breaking change? | yes
| Tickets          | Fix MANAGER-6500
| License          | BSD 3-Clause

### Environment

The `Environment` is no more a Singleton. A new `Environment` is created, hydrated and provided by the `fetchConfiguration`.
Impacts are the following 
* the `ovhManagerCore` module needs the Environment to be defined dynamically, so the AngularJS  module is no more declared by default.
* the ufrontend package, replace the `config` object by the `environment` to inject the `environment` in `application` and `fragments` dynamically.

### Modules

- `@ovh-ux/manager-config` and `@ovh-ux/manager-core` bumped 

### Applications

All applications are up to date and use the `registerApplication` from `@ovh-ux/ufrontend` (EnvExecutor behind), attach/detach the preloader and inject the `environment` in the application. 
Applications are compatible with fragment integration.

Apps
- `@ovh-ux/manager-hub-app`
- `@ovh-ux/manager-dedicated`
- `@ovh-ux/manager-public-cloud`
- `@ovh-ux/manager-web`
- `@ovh-ux/manager-telecom`
- `@ovh-ux/sign-up-app`

Apps using fragment
- `@ovh-ux/manager-freefax-app`
- `fragment/navbar`

Standalone apps
- `@ovh-ux/manager-carrier-sip-app`
- `@ovh-ux/manager-cda-app`
- `@ovh-ux/manager-cloud-connect-app`
- `@ovh-ux/manager-dbaas-logs-app`
- `@ovh-ux/manager-enterprise-cloud-database-app`
- `@ovh-ux/manager-exchange-app`
- `@ovh-ux/manager-iplb-app`
- `@ovh-ux/manager-metrics-app`
- `@ovh-ux/manager-nasha-app`
- `@ovh-ux/manager-office-app`
- `@ovh-ux/order-tracking-app`
- `@ovh-ux/manager-overthebox-app`
- `@ovh-ux/manager-pci-app`
- `@ovh-ux/manager-sharepoint-app`
- `@ovh-ux/manager-sms-app`
- `@ovh-ux/manager-support-app`
- `@ovh-ux/manager-telecom-dashboard-app`
- `@ovh-ux/manager-telecom-task-app`
- `@ovh-ux/manager-veeam-cloud-connect-app`
- `@ovh-ux/manager-veeam-enterprise-app`
- `@ovh-ux/manager-vps-app`
- `@ovh-ux/manager-vrack-app`

### Generator

`@ovh-ux/sao-ovh-manager-app` updated 

### BREAKING CHANGE 💥 

- `@ovh-ux/manager-config` : 
  Environment is no more a singleton
  `fetchConfiguration` returns a new Environment

- `@ovh-ux/manager-core` :
  the core module should be intiliased dynamically
  the `ovhManagerCore` module is no more defined by default
  `registerCoreModule` should be called with environment as first parameter

- `@ovh-ux/ufrontend` : 
  microfrontend exposes now the environment instead of config

### To check

- [x] need to bump `@ovh-ux/ufrontend` (as it's already done in #4504, we have to check if needed when we will merge)
- [x] check ng-app-injection
- [x] squash commmits
- [x] use preloader `displayMessage` in all standalone apps + app generation

### CODE REVIEW

File: packages/manager/apps/exchange/webpack.config.js [hunk 1]
```diff
@@ -44,8 +62,16 @@ module.exports = (env = {}) => {
...
+        /moment[/\\]locale$/,
+        /de|en-gb|es|es-us|fr-ca|fr|it|pl|pt/,
+      ),
+
       new webpack.DefinePlugin({
-        __WEBPACK_REGION__: `'${REGION}'`,
+        __NODE_ENV__: process.env.NODE_ENV
+          ? `'${process.env.NODE_ENV}'`
+          : '"development"',
+        __NG_APP_INJECTIONS__: getNgAppInjections(['EU', 'CA', 'US']),
```
Comment (marie-j) [inline packages/manager/apps/exchange/webpack.config.js:74]:
```suggestion
        __NG_APP_INJECTIONS__: getNgAppInjections(['EU', 'CA']),
```
US needed for Exchange ?

File: packages/manager/apps/dedicated/client/app/app.bundle.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import './app.routes';
+
+import './account/user/advanced/advanced.module';
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply packages/manager/apps/dedicated/client/app/app.bundle.js:3]: based on `develop` / seems to be still there https://github.com/ovh/manager/blob/develop/packages/manager/apps/dedicated/client/app/app.module.js#L62

File: packages/manager/apps/exchange/webpack.config.js (hunk 1 above)
Comment (cbourgois) [reply packages/manager/apps/exchange/webpack.config.js:74]: nope, It's a bad copy/paste. Was done for all apps, so I have to update and check REGION availability (telecom apps...)

Comment (cbourgois) [reply packages/manager/apps/exchange/webpack.config.js:74]: I think we could keep it like this. It's not the app build to decide if the product is enabled or not for `US` region.

Comment (marie-j) [reply packages/manager/apps/exchange/webpack.config.js:74]: @cbourgois Ok but in that cas should we specify a region then ? If it's not the app build responsibility we should be region agnostic, right ?

Comment (cbourgois) [reply packages/manager/apps/exchange/webpack.config.js:74]: Yes, we have to refacto `@ovh-ux/manager-webpack-config` to have all of this in it. Should be done in another PR. Here the goal is to kill Environment singleton.

**Review by antleblanc — APPROVED** [review]

**Pull request merged**